### PR TITLE
Prettying up yaml with setpoint limits, icons, etc...

### PR DESCRIPTION
**Pull request by barndawgie**
I've done some work to clean-up the new YAML, adding default setpoint limits for the climate entities to improve how they render in the frontend as well as default icons for all of the entities without a suitable device class.

### CODE REVIEW

File: econet_heatpump_water_heater.yaml [hunk 1]
```diff
@@ -1,12 +1,19 @@
 substitutions:
   name: "econet-hpwh"
+  friendly_name: "Water Heater"
   device_description: "Rheem Heat Pump Water Heater"
   tx_pin: GPIO19
   rx_pin: GPIO22
+  min_temperature: "43.3333"
+  max_temperature: "60"
```
Comment (tronikos) [inline econet_heatpump_water_heater.yaml]: No need to have substitutions for these. Just inline them.

File: econet_heatpump_water_heater.yaml [hunk 2]
```diff
@@ -30,8 +37,7 @@ logger:
   tx_buffer_size: 2000
 
 external_components:
-  # - source: github://Daniel-dev22/esphome-econet
```
Comment (tronikos) [inline econet_heatpump_water_heater.yaml:33]: Uncomment this and delete your new line

**Comment by tronikos** [comment]
Can you make these changes to https://github.com/stockmopar/esphome-econet instead? You should switch to that repo.

File: econet_hvac.yaml [hunk 3]
```diff
@@ -178,8 +200,10 @@ number:
     max_value: 80
     step: 1
     mode: 'slider'
+    icon: "mdi:air-humidifier-off"
 
 switch:
   - platform: econet
     name: "HVAC Dehumidification Enable State"
     switch_datapoint: DHUMENAB
+    icon: "mdi:air-humidifier-off"
```
Comment (tronikos) [inline econet_hvac.yaml]: Add a new line at the end.

File: econet_heatpump_water_heater.yaml (hunk 1 above)
Comment (barndawgie) [reply econet_heatpump_water_heater.yaml]: Makes sense as these never change, as far as I can tell.

File: econet_heatpump_water_heater.yaml (hunk 2 above)
Comment (barndawgie) [reply econet_heatpump_water_heater.yaml:33]: Not sure I understand - do you just want to keep the local reference or you want something else?

**Comment by barndawgie** [comment]
> Can you make these changes to https://github.com/stockmopar/esphome-econet instead? You should switch to that repo.

Are we moving everything there? Have all of the updates here been merged back up to Sockmopar's repo?

File: econet_heatpump_water_heater.yaml (hunk 2 above)
Comment (tronikos) [reply econet_heatpump_water_heater.yaml:33]: I'm saying the new line you added matches this commented out line. No reason to have both.

**Comment by tronikos** [comment]
> > Can you make these changes to https://github.com/stockmopar/esphome-econet instead? You should switch to that repo.
> 
> Are we moving everything there? Have all of the updates here been merged back up to Sockmopar's repo?

Yes and yes

**Pull request closed without merging**